processing/src/main/java/org/apache/druid/query/rowsandcols/semantic/DefaultFramedOnHeapAggregatable.java

```diff
@@ -75,7 +75,7 @@ public RowsAndColumns aggregateAll(
         int lowerOffset = frame.getLowerOffset();
         int upperOffset = frame.getUpperOffset();
 
-        if (numRows < lowerOffset + upperOffset + 1) {
+        if (numRows < upperOffset - lowerOffset + 1) {
```

Comment:
I think you should also probably take a look at: `WindowFrame#getLowerOffsetClamped`
and the callsites of that method in this file

by the way: I think we should retire the old `rows` processing logic and leave that as well to the one handling the range stuff - it should be on-par in performance; but could handle some edge cases a bit better.